File: src/main/java/no/nav/syfo/controller/domain/sykepengesoknad/oppsummering/Svar.java
package no.nav.syfo.controller.domain.sykepengesoknad.oppsummering;

import java.util.List;

import static java.util.Collections.emptyList;

public class Svar {
    public Ledetekst ledetekst;
    public Svartype type;
    public Tilleggstekst tilleggstekst;
    public List<Sporsmal> undersporsmal = emptyList();

    public Svar withLedetekst(final Ledetekst ledetekst) {
        this.ledetekst = ledetekst;
        return this;
    }

    public Svar withSvartype(final Svartype type) {
        this.type = type;
        return this;
    }

    public Svar withUndersporsmal(final List<Sporsmal> undersporsmal) {
        this.undersporsmal = undersporsmal;
        return this;
    }

    public Svar withTilleggstekst(final Tilleggstekst tilleggstekst) {
        this.tilleggstekst = tilleggstekst;
        return this;
    }
}

File: src/main/kotlin/no/nav/syfo/controller/domain/sykepengesoknad/oppsumering/Svar.kt
package no.nav.syfo.controller.domain.sykepengesoknad.oppsumering

import java.io.Serializable

class Svar : Serializable {
    var ledetekst: Ledetekst? = null
    var type: Svartype? = null
    var tilleggstekst: Tilleggstekst? = null
    var undersporsmal: List<Sporsmal> = emptyList()
    fun withLedetekst(ledetekst: Ledetekst?): Svar {
        this.ledetekst = ledetekst
        return this
    }

    fun withSvartype(type: Svartype?): Svar {
        this.type = type
        return this
    }

    fun withUndersporsmal(undersporsmal: List<Sporsmal>): Svar {
        this.undersporsmal = undersporsmal
        return this
    }

    fun withTilleggstekst(tilleggstekst: Tilleggstekst?): Svar {
        this.tilleggstekst = tilleggstekst
        return this
    }
}
